### PR TITLE
feat: implement toast component

### PR DESCRIPTION
**Pull request by tilucasoli**
### Description

- Create Toast and Dialog component
- Create RemixApp widget it should works like MaterialApp and CupertinoApp

### Changes

- Create Dialog and Toast
- Create ToastLayer
- Create method showDialog and showToast
- Add RemixApp Widget
- Create a function _defineRemixThemeData

**Review Checklist**

- [ ] **Testing**: Have you tested your changes, including unit tests and integration tests for affected code?
- [ ] **Breaking Changes**: Does this change introduce breaking changes affecting existing code or users?
- [ ] **Documentation Updates**: Are all relevant documentation files (e.g. README, API docs) updated to reflect the changes in this PR?
- [ ] **Website Updates**: Is the website containing the updates you make on documentation?

### CODE REVIEW

File: packages/remix/lib/src/components/dialog/dialog.dart [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    transitionDuration: transitionDuration,
+    transitionBuilder: transitionBuilder,
+    useRootNavigator: useRootNavigator,
+    routeSettings: routeSettings,
+    anchorPoint: anchorPoint,
+  );
+}
+
+@MixableSpec()
+base class DialogSpec extends Spec<DialogSpec>
```
Comment (leoafarias) [inline packages/remix/lib/src/components/dialog/dialog.dart:82]: Remove base

**Review by leoafarias — CHANGES_REQUESTED** [review]

File: packages/remix/lib/src/components/dialog/dialog_widget.dart [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+class Dialog extends StatelessWidget {
+  Dialog({
+    super.key,
+    required String title,
+    required String description,
+    this.actions,
+    this.content,
+    this.style,
+    this.variants = const [],
```
Comment (leoafarias) [inline packages/remix/lib/src/components/dialog/dialog_widget.dart:11]: Remove extra parentheses

File: packages/remix/lib/src/components/toast/toast.dart [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  final TextSpec description;
+
+  /// {@macro toast_spec_of}
+  static const of = _$ToastSpec.of;
+
+  static const from = _$ToastSpec.from;
+
+  const ToastSpec({
+    BoxSpec? container,
+    FlexSpec? mainFlex,
```
Comment (leoafarias) [inline packages/remix/lib/src/components/toast/toast.dart:29]: Changing naming to reflect a better reference to the location of the flex

File: packages/remix/lib/src/layout/scaffold.dart [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  const Scaffold({
+    super.key,
+    required this.body,
+    this.color,
+    this.colorOnDark,
+  });
+
+  final Widget body;
+  final Color? color;
+  final Color? colorOnDark;
```
Comment (leoafarias) [inline packages/remix/lib/src/layout/scaffold.dart:17]: Move visual semantics to a ScaffoldStyle

**Review by leoafarias — CHANGES_REQUESTED** [review]

File: packages/remix/lib/src/components/dialog/dialog_widget.dart (hunk 2 above)
Comment (tilucasoli) [reply packages/remix/lib/src/components/dialog/dialog_widget.dart:11]: unfortunately it's needed

**Review by leoafarias — APPROVED** [review]
LGTM

**Pull request merged**